test/adaptive_load/adaptive_load_client_main_test.cc

```diff
@@ -312,10 +313,12 @@ TEST(AdaptiveLoadClientMainTest, WritesOutputProtoToFile) {
   AdaptiveLoadClientMain main(5, argv.data(), controller, filesystem);
   main.Run();
 
-  std::string golden_output =
+  std::string golden_text =
       Envoy::Filesystem::fileSystemForTest().fileReadToEnd(Nighthawk::TestEnvironment::runfilesPath(
           std::string("test/adaptive_load/test_data/golden_output.textproto")));
-  EXPECT_EQ(actual_outfile_contents, golden_output);
+  nighthawk::adaptive_load::AdaptiveLoadSessionOutput golden_proto;
+  Envoy::Protobuf::TextFormat::ParseFromString(golden_text, &golden_proto);
+  EXPECT_EQ(actual_outfile_contents, golden_proto.DebugString());
```

Comment:
I wonder, since we do have an actual proto matcher now, could we compare these two in their binary form instead?

https://github.com/envoyproxy/nighthawk/blob/main/test/test_common/proto_matchers.h

Reply:
The actual value is still a DebugString. Even though current you can deserialize a DebugString just like a textual proto, in the near future this will stop working. The idea here is that we store the textual proto (TextFormat) of the expected proto in the golden file, and create a DebugString of that at run-time. This way no matter how the DebugString format changes (there will be multiple rounds of changes), this test still works.

Reply:
Ah I misunderstood what was going on here, thank you for the explanation.